ptf/sai_base_test.py

```diff
@@ -203,9 +290,38 @@ def get_active_port_list(self):
     def turn_up_and_check_ports(self):
         '''
         Method to turn up the ports.
+        In case some device not init the port after start the switch.
+
+        Args:
+            port_list - list of all active port objects
         '''
         #TODO check if this is common behivor or specified after check on more platform
-        print("For Common platform, Port already setup in recreate_ports.")
+        print("For Common platform, Only check Port status.")
+        
+
+        # For brcm devices, need to init and setup the ports at once after start the switch.
+        retries = 10
+        down_port_list = []
+        for index, oid in enumerate(self.port_list):
+            port_attr = sai_thrift_get_port_attribute(
+                self.client, oid, oper_status=True)
+            print("Turn up port {}".format(oid))
+            port_up = True
+            if port_attr['oper_status'] != SAI_PORT_OPER_STATUS_UP:
+                port_up = False
+                for num_of_tries in range(retries):
+                    port_attr = sai_thrift_get_port_attribute(
+                        self.client, oid, oper_status=True)
+                    if port_attr['oper_status'] == SAI_PORT_OPER_STATUS_UP:
+                        port_up = True
+                        break
+                    time.sleep(5)
```

Comment:
@ravi861 please explain this sleeps if they are needed, they slow down out tests, waiting and retrying all time

Reply:
Although the changes came in via this PR, this specific timeout was initiated by Richard during his testing with BRCM devices as the comment says above.
Typically, the test would not have to retry so many times if the links come up fast enough. An alternative to hard-coding 5 sec is to pass in a test param and make it configurable.

Reply:
in the DASH repo scenario where those tests are played, they are played gains p4/bmv2 switch, and those ports are never get up, at least what i observed notifications are not supported yet, so it waits 5x10 - 50 seconds to retry, and this is not the only tests, there are couple of them

Reply:
you can see the retrying here: https://github.com/sonic-net/DASH/actions/runs/6053205361/job/16428197918?pr=432#step:17:80